unicodetools/org/unicode/utilities/UnicodeSetParser.java

```diff
@@ -0,0 +1,881 @@
+package org.unicode.utilities;
+
+import java.io.IOException;
+import java.text.ParsePosition;
+import java.util.Set;
+import java.util.TreeSet;
+
+import com.ibm.icu.impl.PatternProps;
+import com.ibm.icu.impl.RuleCharacterIterator;
+import com.ibm.icu.impl.StringRange;
+import com.ibm.icu.impl.Utility;
+import com.ibm.icu.lang.CharSequences;
+import com.ibm.icu.text.SymbolTable;
+import com.ibm.icu.text.UTF16;
+import com.ibm.icu.text.UnicodeMatcher;
+import com.ibm.icu.text.UnicodeSet;
+import com.ibm.icu.util.ICUUncheckedIOException;
+
+public class UnicodeSetParser {
```

Comment:
What is the purpose of this?
Is it a copy of the parser inside ICU's UnicodeSet?
If so, why make a copy?
What does it do differently?